webpack.config.js

```diff
@@ -0,0 +1,245 @@
+						{
+							// loader: 'typings-for-css-modules-loader?modules&localIdentName=[local]__[hash:base64:5]&importLoaders=1'+(isProd ? '&sourceMap' : '')
+							// loader: 'css-loader',
+							loader: 'typings-for-css-modules-loader',
+							options: {
+								modules: true,
+								localIdentName: '[local]__[hash:base64:5]',
+								namedExport: true,
+								camelCase: true,
+								importLoaders: 1,
+								sourceMap: isProd,
+								sass: true
+							}
+						}
+					]
+					// loader: ExtractTextPlugin.extract({
+					// 	fallback: 'style-loader',
+					// 	use: [{
+					// 		loader: 'css-loader',
+					// 		options: {
+					// 			modules: true,
+					// 			localIdentName: '[local]__[hash:base64:5]',
+					// 			importLoaders: 1,
+					// 			sourceMap: isProd
+					// 		}
+					// 	}]
+					// })
+				},
+				{
+					test: /\.(scss|sass|css)$/,
+					exclude: componentStyleDirs,
+					use: [
+						isProd ? MiniCssExtractPlugin.loader : 'style-loader',
+						{
+							loader: 'css-loader',
+							options: {
+								importLoaders: 1,
+								sourceMap: isProd
+							}
+						}
+					]
+					// loader: ExtractTextPlugin.extract({
+					// 	fallback: 'style-loader',
+					// 	use: [{
+					// 		loader: 'css-loader',
+					// 		options: {
+					// 			importLoaders: 1,
+					// 			sourceMap: isProd
+					// 		}
+					// 	}]
+					// })
+				}
+			]
+		},
+		plugins: [
+			new ProgressBarPlugin({
+				format: '\u001b[90m\u001b[44mBuild\u001b[49m\u001b[39m [:bar] \u001b[32m\u001b[1m:percent\u001b[22m\u001b[39m (:elapseds) \u001b[2m:msg\u001b[22m',
+				renderThrottle: 100,
+				summary: false,
+				clear: true
+			}),
+
+			isProd && new CleanWebpackPlugin(['build']),
+			isProd && new webpack.optimize.SplitChunksPlugin({}),
+			isProd && new MiniCssExtractPlugin({}),
+			// new ExtractTextPlugin({
+			// 	filename: isProd ? 'style.[contenthash:5].css' : 'style.css',
+			// 	disable: !isProd,
+			// 	allChunks: true
+			// }),
+
+			// fixes infinite loop in typings-for-css-modules-loader:
+			new webpack.WatchIgnorePlugin([
+				/(c|sc|sa)ss\.d\.ts$/
+			]),
+			new WatchTimestampsPlugin([
+				/(c|sc|sa)ss\.d\.ts$/
+			]),
+			new HtmlWebpackPlugin({
+				filename: path.join(__dirname, 'build/index.html'),
+				template: '!!ejs-loader!src/index.html',
+				minify: isProd && {
+					collapseWhitespace: true,
+					removeScriptTypeAttributes: true,
+					removeRedundantAttributes: true,
+					removeStyleLinkTypeAttributes: true,
+					removeComments: true
+				},
+				manifest,
+
+				/** @todo Finish implementing prerendering similar to that of Preact CLI. */
+				prerender() {
+					return '<div id="app_root"></div>';
+					// require('babel-register')({ ignore: false });
+					// return require('./config/prerender')();
+				},
+				inject: true,
+				compile: true
+			}),
+			isProd && new PreloadWebpackPlugin(),
+			new webpack.DefinePlugin({
+				process: '{}'
+			}),
+			isProd && new ReplacePlugin({
```

Comment:
What does this do? Are we suppressing throws in prod here?

Reply:
It strips out Babel class helper code that throws if classes are invoked without `new`, which is unnecessary for a production build, but UglifyJS does not statically remove it. If we migrate to Babel 7 (beta), this may no longer be necessary since `loose` mode omits these helpers.